### PR TITLE
Various optimizations

### PR DESCRIPTION
**Pull request by nibraswastaken**
This PR optimizes nearly everything, notable changes:

- Remove unused sources
- Split and move meta tags to respective documents (`_app.tsx` and `_document.tsx`)
- Use `getStaticProps` instead of `getInitialProps`
- Use `cross-fetch` instead of `node-fetch`
- ~~Adding `unstable_runtimeJS`~~
- Enable experimental features
- Use preact in production

### CODE REVIEW

**Comment by nibraswastaken** [comment]
Note the build logs below, fist load JS is greatly reduced and build time is slightly reduced.

<details>
<summary>Local build log from current main branch</summary>
Screen Shot 2021-07-07 at 7 16 57 PM
</details>

<details>
<summary>Local build log from PR</summary>
Screen Shot 2021-07-07 at 7 18 00 PM
</details>

File: pages/_document.tsx [hunk 1]
```diff
@@ -1,10 +1,31 @@
...
+  Main,
+  NextScript,
+  DocumentContext,
+} from "next/document";
+
+export default class MyDocument extends Document {
+  static async getInitialProps(ctx: DocumentContext) {
+    const initialProps = await Document.getInitialProps(ctx);
+    return { ...initialProps };
+  }
```
Comment (zainfathoni) [inline pages/_document.tsx:13]:
```suggestion
```

I intentionally left this out because I have no intention of changing its behaviour.

Screenshot 2021-07-08 at 00 37 06

File: etc/fetchers/fetch-wbw.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 const cheerio = require("cheerio");
 const fs = require("fs");
 const path = require("path");
-const fetch = require("node-fetch");
+const { fetch } = require("cross-fetch");
```
Comment (zainfathoni) [inline etc/fetchers/fetch-wbw.js:4]: What's the benefit of this `cross-fetch` over `node-fetch`?

File: next.config.js [hunk 3]
```diff
@@ -1,3 +1,27 @@
...
+    if (!dev && !isServer) {
+      Object.assign(config.resolve.alias, {
+        react: "preact/compat",
+        "react-dom/test-utils": "preact/test-utils",
+        "react-dom": "preact/compat",
+      });
+    }
+
+    return config;
+  },
```
Comment (zainfathoni) [inline next.config.js:26]: I'm hesitant about the effectiveness of these changes.
If it turns out damaging the performance, let's keep these improvements as a separate PR.

File: package.json [hunk 4]
```diff
@@ -31,7 +32,15 @@
...
-  }
+    "*.{js,json,md}": [
+      "prettier --write"
+    ],
+    "**/**.css": [
+      "prettier --write"
+    ],
+    "**/*.{js,jsx,ts,tsx}": [
+      "eslint --fix"
+    ]
```
Comment (zainfathoni) [inline package.json:43]: Why are you avoiding prettier from formating the TS files?
I thought ESLint and Prettier should work well side-by-side with the current setup.

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Let's wait for #20 to be merged first to measure its impact on the performance objectively.

**Comment by zainfathoni** [comment]
Screenshot 2021-07-09 at 22 22 38

I think I know what made your deployment preview's Lighthouse scores are so bad.

That's because in the deployment previews, Netlify injected a script to provide an interactive collaboration feature. 😅

**Comment by zainfathoni** [comment]
> I think I know what made your deployment preview's Lighthouse scores are so bad.
> 
> That's because, in the deployment previews, Netlify injected a script to provide an interactive collaboration feature. 😅

After rerunning the measurement on the pristine URL (without Netlify-injected functionalities), the scores are actually pretty good. 😅 

Screenshot 2021-07-12 at 08 28 13

File: etc/fetchers/fetch-wbw.js (hunk 2 above)
Comment (resir014) [reply etc/fetchers/fetch-wbw.js:4]: @zainfathoni Works on both server-side and client-side.

File: next.config.js (hunk 3 above)
Comment (resir014) [reply next.config.js:26]: We can go ahead with the experimental stuff and strict mode. For preact, let's hold off on that until the project is stable enough.

File: package.json (hunk 4 above)
Comment (resir014) [reply package.json:43]: If we include `eslint-plugin-prettier` in the ESLint config, it should automatically take all config rules + autofixes directly from the Prettier config, without having to go through Prettier.

**Review by resir014 — APPROVED** [review]
